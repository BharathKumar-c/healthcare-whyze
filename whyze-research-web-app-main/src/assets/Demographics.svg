<svg width="40" height="40" viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg">
<rect width="40" height="40" rx="20" fill="#FFF7E0"/>
<path d="M20.375 29.2242C19.925 29.2617 19.5125 29.2617 19.1 29.2617C15.9125 29.2617 12.875 28.0242 11.6 27.4242C11.7125 24.9867 12.5 22.8117 13.85 21.2367C15.2375 19.5867 17.1125 18.6867 19.1 18.6867C20.6375 18.6867 22.0625 19.2117 23.3 20.1867C23.6 20.4117 24.0125 20.3742 24.2375 20.0742C24.4625 19.7742 24.425 19.3617 24.125 19.1367C23.15 18.3492 22.0625 17.8242 20.9 17.5617C22.625 16.8492 23.8625 15.1617 23.8625 13.1742C23.8625 10.5492 21.725 8.44922 19.1375 8.44922C16.5125 8.44922 14.4125 10.5867 14.4125 13.1742C14.4125 15.1617 15.6125 16.8492 17.3375 17.5617C15.65 17.9742 14.1125 18.9117 12.875 20.3742C11.225 22.2867 10.325 24.9492 10.25 27.8367C10.25 28.0992 10.4 28.3242 10.625 28.4367C11.6375 28.9242 15.2375 30.5742 19.1 30.5742C19.55 30.5742 20.0375 30.5367 20.4875 30.4992C20.8625 30.4617 21.125 30.1617 21.0875 29.7867C21.05 29.4492 20.75 29.1867 20.375 29.2242ZM15.65 13.2117C15.65 11.3367 17.1875 9.79922 19.0625 9.79922C20.9375 9.79922 22.475 11.3367 22.475 13.2117C22.475 15.0867 20.9375 16.6242 19.0625 16.6242C17.1875 16.6242 15.65 15.0867 15.65 13.2117Z" fill="#FABB00"/>
<path d="M29.6374 30.4988L28.0999 28.3613C28.7374 27.7238 29.1499 26.9363 29.2624 26.0738C29.4499 24.9863 29.1874 23.8613 28.5124 22.9613C27.8749 22.0613 26.8999 21.4613 25.7749 21.2738C24.6874 21.0863 23.5624 21.3488 22.6624 22.0238C20.7874 23.3738 20.3749 25.9988 21.7249 27.8738C22.5499 28.9988 23.8249 29.5988 25.1374 29.5988C25.7749 29.5988 26.4124 29.4488 27.0124 29.1488L28.5499 31.2863C28.6624 31.4738 28.8874 31.5488 29.0749 31.5488C29.2249 31.5488 29.3374 31.5113 29.4499 31.4363C29.7499 31.2113 29.8249 30.7988 29.6374 30.4988ZM22.7749 27.0863C22.3249 26.4488 22.1374 25.6988 22.2499 24.9488C22.3624 24.1988 22.7749 23.5238 23.4124 23.0738C23.8999 22.6988 24.4999 22.5113 25.0999 22.5113C25.2499 22.5113 25.3999 22.5113 25.5499 22.5488C26.2999 22.6613 26.9749 23.0738 27.4249 23.7113C27.8749 24.3488 28.0624 25.0988 27.9499 25.8488C27.8374 26.5988 27.4249 27.2738 26.7874 27.7238C25.5124 28.6613 23.7124 28.3613 22.7749 27.0863Z" fill="#FABB00"/>
</svg>
