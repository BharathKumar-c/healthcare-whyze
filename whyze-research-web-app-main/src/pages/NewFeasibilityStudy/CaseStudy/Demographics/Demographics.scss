@import '../../../../common_styles/common_variable.scss';

.demographics_container {

   display: flex;
   flex-direction: column;
   gap: 2rem;

   .demographics_gender-wrapper {
      display: flex;
      flex-direction: row;
      align-items: center;

      &_drag-icon {
         margin-left: -1rem;
      }

      &_checkbox {
         display: flex !important;
         flex-direction: row !important;
         margin-left: -2.25rem;
      }

      &_gender-remove-btn {
         cursor: pointer;
         align-items: flex-end;
         display: flex;
         justify-content: center;
         margin-left: -2rem;
      }

      &_selected-noof-patients {
         float: right !important;
         margin-bottom: -0.5rem;
      }

      &_totalpatient-tag {
         display: flex !important;
         flex-direction: row;
         justify-content: flex-end;
         align-items: center;
         padding: 4px 8px !important;
         max-width: max-content;
         background: #f6fefe !important;
         border: 1px solid #d2f9f9 !important;
         border-radius: 16px !important;
         font-style: normal;
         font-weight: 700;
         font-size: 12px !important;
         line-height: 16px !important;
         color: $secondary-text-color !important;
      }
   }

   .demographics_agerange-wrapper {
      display: flex;
      flex-direction: row;
      align-items: center;

      &_drag-icon {
         margin-left: -1rem;
         margin-top: 1.5rem;
      }

      &_Input-box1 {
         margin-top: 1.4rem;
      }

      @media screen and (min-width: 1600px) {

         &_Input-box {
            max-width: none;
            padding-right: 1.3rem;
         }
      }

      @media screen and (min-width: 756px) {

         &_Input-box {
            max-width: none;
            padding-right: 1.3rem;
         }
      }

      &_ageRange-remove-btn {
         cursor: pointer;
         align-items: flex-end;
         display: flex;
         justify-content: center;
         margin-top: 1rem;
      }

      &_selected-noof-patients {
         float: right !important;
         margin-bottom: -0.5rem;
      }

      &_totalpatient-tag {
         display: flex !important;
         flex-direction: row;
         justify-content: flex-end;
         align-items: center;
         padding: 4px 8px !important;
         max-width: max-content;
         background: #f6fefe !important;
         border: 1px solid #d2f9f9 !important;
         border-radius: 16px !important;
         font-style: normal;
         font-weight: 700;
         font-size: 12px !important;
         line-height: 16px !important;
         color: $secondary-text-color !important;
      }
   }

   h4 {
      margin: 0rem 0rem 0rem 2rem;
      font-size: 15px;
      font-weight: 900;
   }
}