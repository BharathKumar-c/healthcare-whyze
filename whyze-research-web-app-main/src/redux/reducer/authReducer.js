import { createSlice } from '@reduxjs/toolkit';
import jwt_decode from 'jwt-decode';

const initialState = {
  isLoggedIn: loginCheck(),
  accessToken: null,
  refreshToken: null,
  isLoading: false,
  isDisableForgot: false,
  message: '',
  userDetails: {},
  profileCount: 0,
};

function loginCheck() {
  const credentials = JSON.parse(localStorage.getItem('credentials'));
  if (credentials?.access_token) {
    const { exp } = jwt_decode(credentials.access_token);
    if (Date.now() <= exp * 1000) {
      return true;
    }
  }
  return false;
}

const authSlice = createSlice({
  name: 'authAction',
  initialState: initialState,
  reducers: {
    setLoginStatus: (state, { payload }) => {
      state.isLoggedIn = payload;
    },
    setAccessToken: (state, { payload }) => {
      state.accessToken = payload;
    },
    setRefreshToken: (state, { payload }) => {
      state.refreshToken = payload;
    },
    setLoading: (state, { payload }) => {
      state.isLoading = payload;
    },
    setDisableForgot: (state, { payload }) => {
      state.isDisableForgot = payload;
    },
    setResetMessage: (state, { payload }) => {
      state.message = payload;
    },
    setUserDetails: (state, { payload }) => {
      state.userDetails = payload;
    },
    setProfileCount: (state) => {
      state.profileCount = state.profileCount + 1;
    },
  },
});

export const {
  setLoginStatus,
  setAccessToken,
  setRefreshToken,
  setLoading,
  setDisableForgot,
  setResetMessage,
  setUserDetails,
  setProfileCount,
} = authSlice.actions;

export default authSlice.reducer;
