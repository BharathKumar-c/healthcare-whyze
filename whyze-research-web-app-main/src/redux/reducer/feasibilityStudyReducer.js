import { createSlice } from '@reduxjs/toolkit';
import { v4 as uuid } from 'uuid';

export const initialState = {
  segments: [
    {
      name: 'Segment-1',
      isDone: false,
      activeTab: {
        name: 'feasibility',
        index: 0,
      },
      tabs: [
        {
          name: 'feasibility',
          title: 'Feasibility',
          isActive: true,
          position: 'left',
        },
      ],
      feasibilityStudy: [
        {
          name: 'Indication',
          case: 'Indication',
          condition: 'Inclusion',
          caseName: '',
          child: [],
        },
        {
          name: 'Demographics',
          case: 'Demographics',
          condition: 'Inclusion',
          caseName: '',
          child: [
            {
              id: uuid().slice(0, 8),
              condition: 'Gender',
              field: 'gender',
              gender: '',
            },
            {
              id: uuid().slice(0, 8),
              condition: 'Age range',
              field: 'age',
              ageRange: {
                min: 0,
                max: 0,
              },
            },
          ],
        },
      ],
      country: [],
      location: [],
      selectedCountries: [],
      totalConnectedNotConnected: {},
      sites: [],
      sitesContacted: [],
      sitesInitiated: [],
      preferredSites: [],
      patientCount: {
        feasibility: 0,
        location: 0,
      },
      siteCount: {
        feasibility: 0,
        location: 0,
      },
    },
  ],
  cardValue: {
    case: 'Indication',
    condition: '',
    index: 0,
  },
  activeSegment: {
    name: '',
    index: 0,
  },
};

const feasibilityStudySlice = createSlice({
  name: 'feasibilityStudyAction',
  initialState: initialState,
  reducers: {
    setCardDetails: (state, { payload }) => {
      state.cardValue = payload;
    },
    setSegments: (state, { payload }) => {
      state.segments = payload;
    },
    setActiveSegment: (state, { payload }) => {
      state.activeSegment = payload;
    },
    resetSegments: (state) => {
      state.segments = initialState.segments;
      state.cardValue = {
        case: 'Indication',
        condition: '',
        index: 0,
      };
      state.activeSegment = {
        name: '',
        index: 0,
      };
    },
  },
});

export const { setCardDetails, setSegments, setActiveSegment, resetSegments } =
  feasibilityStudySlice.actions;

export default feasibilityStudySlice.reducer;
