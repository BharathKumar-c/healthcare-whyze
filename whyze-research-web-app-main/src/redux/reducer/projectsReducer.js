import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  projects: [],
  projectId: '',
  clinicalTrial: '',
  sponsor: '',
  description: '',
  measurement: 'Metric',
  standards: '',
  segmentId: [],
};

const projectSlice = createSlice({
  name: 'projectAction',
  initialState: initialState,
  reducers: {
    setProjects: (state, { payload }) => {
      state.projects = payload;
    },
    setProjectId: (state, { payload }) => {
      state.projectId = payload;
    },
    setClinicalTrial: (state, { payload }) => {
      state.clinicalTrial = payload;
    },
    setSponsor: (state, { payload }) => {
      state.sponsor = payload;
    },
    setDescription: (state, { payload }) => {
      state.description = payload;
    },
    setStandards: (state, { payload }) => {
      state.standards = payload;
    },
    setMeasurement: (state, { payload }) => {
      state.measurement = payload;
    },
    setSegmentId: (state, { payload }) => {
      state.segmentId = payload;
    },
    setProjectSettings: (state, { payload }) => {
      state.projectId = payload.projectId;
      state.segmentId = payload.segmentId;
      state.clinicalTrial = payload.clinicalTrial;
      state.sponsor = payload.sponsor;
      state.description = payload.description;
      state.measurement = payload.measurement;
      state.standards = payload.standards;
    },
    clearSettings: (state) => {
      state.clinicalTrial = '';
      state.sponsor = '';
      state.description = '';
      state.measurement = 'Metric';
      state.standards = '';
      state.projectId = '';
      state.segmentId = [];
    },
  },
});

export const {
  setProjects,
  setProjectId,
  setProjectSettings,
  setClinicalTrial,
  setDescription,
  setSponsor,
  setMeasurement,
  setStandards,
  setSegmentId,
  clearSettings,
} = projectSlice.actions;

export default projectSlice.reducer;
