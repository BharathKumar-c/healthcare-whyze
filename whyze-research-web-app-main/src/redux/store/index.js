import { configureStore } from '@reduxjs/toolkit';
import reducer from '../reducer/combinedReducer';

const store = configureStore({
  reducer: reducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
