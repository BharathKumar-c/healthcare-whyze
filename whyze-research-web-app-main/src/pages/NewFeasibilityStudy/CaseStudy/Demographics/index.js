import React from 'react';
import { Row, Col } from 'antd';
import './Demographics.scss';
import PropTypes from 'prop-types';
import { Drag } from '../../../../assets';
import { Vector2 } from '../../../../assets';
import { totalPatientFormatter } from '../../../../utils/calculations';
import { ConditionContainer } from '../../../../components/shared';
import { Gender } from '../../../../constant';
import { Checkbox, Tag, Input } from '../../../../components/basic';
import { useDispatch, useSelector } from 'react-redux';
import { setSegments } from '../../../../redux/reducer/feasibilityStudyReducer';
import { demographicsConst } from '../../../../constant/ConstantTexts';
import useNewFeasibilityStudyService from '../../useNewFeasibilityStudyService';

export default function Demographics({
  currentIndex,
  caseDetails,
  type,
  label,
}) {
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const dispatch = useDispatch();
  const { optimusValue, isDisableInput } = useNewFeasibilityStudyService();

  const handleSelectGender = (data) => {
    const study = JSON.parse(JSON.stringify(segments));
    const { value, mainIndex, subIndex } = data;
    const gender =
      study[activeSegment.index]?.feasibilityStudy[mainIndex]?.child[subIndex]
        ?.gender;
    const newGender = gender
      ? gender?.includes(',')
        ? gender?.split(',')
        : [gender]
      : [];
    let edittedGender = [...newGender];
    if (newGender.includes(value.target.name)) {
      edittedGender = newGender.filter((ele) => ele !== value.target.name);
    } else {
      edittedGender.push(value.target.name);
    }
    study[activeSegment.index].feasibilityStudy[mainIndex].child[
      subIndex
    ].gender = edittedGender.join();
    optimusValue({
      data: study[activeSegment.index]?.feasibilityStudy,
      id: study[activeSegment.index].feasibilityStudy[mainIndex].child[subIndex]
        .id,
    });
    dispatch(setSegments(study));
  };

  const handleInputChange = ({ e, mainIndex, subIndex }) => {
    const study = JSON.parse(JSON.stringify(segments));
    study[activeSegment.index].feasibilityStudy[mainIndex].child[
      subIndex
    ].ageRange = {
      ...study[activeSegment.index]?.feasibilityStudy[mainIndex]?.child[
        subIndex
      ]?.ageRange,
      [e.target.name]: e.target.value,
    };
    optimusValue({
      data: study[activeSegment.index]?.feasibilityStudy,
      id: study[activeSegment.index].feasibilityStudy[mainIndex].child[subIndex]
        .id,
    });
    dispatch(setSegments(study));
  };

  return (
    <div className="demographics_container">
      {caseDetails.map((c, index) => (
        <ConditionContainer
          className={'condition-container'}
          key={index}
          isActive
          isVertical
        >
          {c.field === demographicsConst.age && (
            <>
              <Row
                gutter={{ sm: 32, md: 32, lg: 32 }}
                className="demographics_agerange-wrapper"
              >
                <Col
                  className="demographics_agerange-wrapper_drag-icon"
                  md={2}
                  sm={1}
                  lg={1}
                  xl={1}
                >
                  <Drag />
                </Col>
                <Col sm={19} md={17} lg={14} xl={14} xxl={14}>
                  <Row>
                    <Col
                      xl={6}
                      sm={8}
                      md={10}
                      lg={9}
                      xxl={3}
                      className="demographics_agerange-wrapper_Input-box"
                    >
                      <Input
                        type={type}
                        lable={label}
                        width={100}
                        name={demographicsConst.min}
                        value={c?.ageRange?.min}
                        handleInputChange={(e) =>
                          handleInputChange({
                            e,
                            mainIndex: currentIndex,
                            subIndex: index,
                          })
                        }
                        disabled={isDisableInput}
                        onWheel={(e) => e.target.blur()}
                      />
                    </Col>
                    <Col className="demographics_agerange-wrapper_Input-box1">
                      <Input
                        type={type}
                        width={100}
                        name={demographicsConst.max}
                        value={c?.ageRange?.max}
                        handleInputChange={(e) =>
                          handleInputChange({
                            e,
                            mainIndex: currentIndex,
                            subIndex: index,
                          })
                        }
                        disabled={isDisableInput}
                        onWheel={(e) => e.target.blur()}
                      />
                    </Col>
                  </Row>
                </Col>
              </Row>
              <Row className="demographics_agerange-wrapper_selected-noof-patients">
                <Col>
                  <Tag
                    icon={<Vector2 />}
                    className="demographics_agerange-wrapper_totalpatient-tag"
                  >
                    {totalPatientFormatter(
                      segments[activeSegment.index]?.feasibilityStudy[
                        currentIndex
                      ]?.child[index].count || '0000',
                    )}
                  </Tag>
                </Col>
              </Row>
            </>
          )}

          {c.field === demographicsConst.gender && (
            <>
              <Row>
                <h4>Sex</h4>
              </Row>
              <Row
                gutter={{ sm: 32, md: 32, lg: 32 }}
                className="demographics_gender-wrapper"
              >
                <Col
                  className="demographics_gender-wrapper_drag-icon"
                  md={4}
                  lg={4}
                  xl={2}
                >
                  <Drag />
                </Col>

                <Col
                  sm={19}
                  md={17}
                  lg={14}
                  xl={14}
                  className="demographics_gender-wrapper_checkbox"
                >
                  {Gender.map((ele, i) => (
                    <Checkbox
                      className={
                        c?.gender?.includes(ele.gender)
                          ? 'input-group-active'
                          : 'input-group'
                      }
                      key={i}
                      name={ele.gender}
                      lable={ele.gender}
                      value={c?.gender?.includes(ele.gender)}
                      handleChecked={(value) => {
                        handleSelectGender({
                          value,
                          mainIndex: currentIndex,
                          subIndex: index,
                        });
                      }}
                    />
                  ))}
                </Col>
              </Row>
              <Row className="demographics_gender-wrapper_selected-noof-patients">
                <Col>
                  <Tag
                    icon={<Vector2 />}
                    className="demographics_gender-wrapper_totalpatient-tag"
                  >
                    {totalPatientFormatter(
                      segments[activeSegment.index]?.feasibilityStudy[
                        currentIndex
                      ]?.child[index].count || '0000',
                    )}
                  </Tag>
                </Col>
              </Row>
            </>
          )}
        </ConditionContainer>
      ))}
    </div>
  );
}

Demographics.propTypes = {
  caseDetails: PropTypes.array,
  currentIndex: PropTypes.number,
  type: PropTypes.string,
  label: PropTypes.string,
};
Demographics.defalutProps = {
  caseDetails: [],
  currentIndex: 0,
  type: '',
  label: '',
};
