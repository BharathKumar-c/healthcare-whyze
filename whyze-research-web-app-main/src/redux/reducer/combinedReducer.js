import { combineReducers } from 'redux';
import feasibilityStudy from './feasibilityStudyReducer';
import project from './projectsReducer';
import auth from './authReducer';

export default combineReducers({
  feasibilityStudyReducer: feasibilityStudy,
  authReducer: auth,
  projectReducer: project,
});
