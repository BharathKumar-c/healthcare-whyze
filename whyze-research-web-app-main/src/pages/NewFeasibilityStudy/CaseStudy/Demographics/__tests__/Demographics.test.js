import React from 'react';
import { act, fireEvent, render, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import { setSegments } from '../../../../../redux/reducer/feasibilityStudyReducer';
import store from '../../../../../redux/store';
import Demographics from '..';
const mockedUsedNavigate = jest.fn();
import ApiUtil from '../../../../../utils/ApiUtils';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

jest.mock('axios');
jest.mock('../../../../../utils/ApiUtils');

function renderWithContext(element) {
  const { container, getByText } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByText };
}
const caseDetails = [
  {
    id: '07b8d849',
    condition: 'Gender',
    field: 'gender',
    gender: '',
  },
  {
    id: '06b35f26',
    condition: 'Age range',
    field: 'age',
    ageRange: {
      min: 0,
      max: 0,
    },
  },
];

describe('Test cases for Demographics', () => {
  test('Create the snapshots for Demographics', () => {
    const { container } = renderWithContext(
      <Demographics caseDetails={caseDetails} />,
    );

    act(() => {
      const { segments, activeSegment } = JSON.parse(
        JSON.stringify(store.getState().feasibilityStudyReducer),
      );
      segments[activeSegment.index].feasibilityStudy.push(caseDetails);
      store.dispatch(setSegments(segments));
    });

    expect(container).toMatchSnapshot();
  });

  test('Should render the input field correctly', async () => {
    const { container } = renderWithContext(
      <Demographics caseDetails={caseDetails} />,
    );
    const minAgeRangeInputField = container.querySelector('input[name="min"]');
    expect(minAgeRangeInputField).toBeInTheDocument();
    const maxAgeRangeInputField = container.querySelector('input[name="max"]');
    expect(maxAgeRangeInputField).toBeInTheDocument();
    const maleInputField = container.querySelector('input[name="Male"]');
    expect(maleInputField).toBeInTheDocument();
    const femaleInputField = container.querySelector('input[name="Female"]');
    expect(femaleInputField).toBeInTheDocument();
  });

  test('Should render the No of patient tag correctly when sent the total patients count', () => {
    const { container } = renderWithContext(
      <Demographics caseDetails={caseDetails} totalpatient={10000} />,
    );
    const noOfPatinetsAgeRangeTag = container.getElementsByClassName(
      'demographics_agerange-wrapper_totalpatient-tag',
    );
    expect(noOfPatinetsAgeRangeTag[0]).toBeInTheDocument();
    const noOfPatinetsGenderTag = container.getElementsByClassName(
      'demographics_gender-wrapper_totalpatient-tag',
    );
    expect(noOfPatinetsGenderTag[0]).toBeInTheDocument();
  });

  test('Test on change event', async () => {
    const { container } = renderWithContext(
      <Demographics
        caseDetails={caseDetails}
        currentIndex={1}
        totalpatient={10000}
      />,
    );

    const minAgeRangeInputField = container.querySelector('input[name="min"]');
    expect(minAgeRangeInputField).toBeInTheDocument();
    const maxAgeRangeInputField = container.querySelector('input[name="max"]');
    expect(maxAgeRangeInputField).toBeInTheDocument();
    const maleInputField = container.querySelector('input[name="Male"]');
    expect(maleInputField).toBeInTheDocument();
    const femaleInputField = container.querySelector('input[name="Female"]');
    expect(femaleInputField).toBeInTheDocument();

    fireEvent.change(minAgeRangeInputField, {
      target: { value: '10' },
    });
    fireEvent.change(maxAgeRangeInputField, {
      target: { value: '50' },
    });
    fireEvent.click(maleInputField);
    fireEvent.click(femaleInputField);

    await waitFor(() => {
      const { segments, activeSegment } =
        store.getState().feasibilityStudyReducer;

      expect(
        segments[activeSegment.index]?.feasibilityStudy[1].child[1].ageRange
          .min,
      ).toBe('10');
      expect(
        segments[activeSegment.index]?.feasibilityStudy[1].child[1].ageRange
          .max,
      ).toBe('50');
      expect(
        segments[activeSegment.index]?.feasibilityStudy[1].child[0].gender,
      ).toBe('Male,Female');
      expect(
        segments[activeSegment.index]?.feasibilityStudy[1].child[0].gender,
      ).toBe('Male,Female');
    });
  });

  test('API Get Count ', async () => {
    renderWithContext(
      <Demographics
        caseDetails={caseDetails}
        currentIndex={1}
        totalpatient={10000}
      />,
    );
    ApiUtil.postData.mockResolvedValue({
      data: {
        caseStudy: [
          {
            name: 'Indication',
            case: 'Indication',
            condition: 'Inclusion',
            caseName: 'Covid-19',
            child: [
              {
                id: 'e391a071',
                condition: 'diagnosed',
                operation: 'And',
                child: [],
                diagnosed: {
                  period: 'day',
                  value: '22',
                },
                count: 3984,
              },
            ],
          },
          {
            name: 'Demographics',
            case: 'Demographics',
            condition: 'Inclusion',
            caseName: '',
            child: [
              {
                id: '486ed60d',
                condition: 'Gender',
                field: 'gender',
                gender: 'Male',
                count: 3150,
              },
              {
                id: 'fbca0dfa',
                condition: 'Age range',
                field: 'age',
                ageRange: {
                  min: 0,
                  max: 0,
                },
              },
            ],
          },
        ],
        country: [
          {
            Country: 'United Kingdom',
            count: '1',
          },
          {
            Country: 'Albania',
            count: '1',
          },
          {
            Country: null,
            count: '3137',
          },
          {
            Country: 'Brazil',
            count: '4',
          },
          {
            Country: 'Ireland',
            count: '4',
          },
          {
            Country: 'Portugal',
            count: '3',
          },
        ],
      },
      status: 200,
      statusText: '',
      headers: {
        'content-length': '724',
        'content-type': 'application/json; charset=utf-8',
      },
      request: {},
    });
    await waitFor(() => {
      expect(ApiUtil.postData).toHaveBeenCalledTimes(1);
    });
  });
});
