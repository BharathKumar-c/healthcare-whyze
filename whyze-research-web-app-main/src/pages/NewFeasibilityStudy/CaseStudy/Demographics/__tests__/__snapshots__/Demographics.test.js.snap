// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases for Demographics Create the snapshots for Demographics 1`] = `
<div>
  <div
    class="demographics_container"
  >
    <div
      class="ant-space ant-space-horizontal ant-space-align-center condition-container"
      style="border: 2px solid #d3d6d9; display: flex; flex-direction: column; margin-bottom: 0.5rem; padding: 20px 30px 12px 20px;"
    >
      <div
        class="ant-space-item"
        style="margin-right: 8px;"
      >
        <div
          class="ant-row"
        >
          <h4>
            Sex
          </h4>
        </div>
      </div>
      <div
        class="ant-space-item"
        style="margin-right: 8px;"
      >
        <div
          class="ant-row demographics_gender-wrapper"
          style="margin-left: -16px; margin-right: -16px;"
        >
          <div
            class="ant-col demographics_gender-wrapper_drag-icon ant-col-md-4 ant-col-lg-4 ant-col-xl-2"
            style="padding-left: 16px; padding-right: 16px;"
          >
            <svg>
              Drag.svg
            </svg>
          </div>
          <div
            class="ant-col demographics_gender-wrapper_checkbox ant-col-sm-19 ant-col-md-17 ant-col-lg-14 ant-col-xl-14"
            style="padding-left: 16px; padding-right: 16px;"
          >
            <div
              class="input-group"
            >
              <label
                class="ant-checkbox-wrapper"
              >
                <span
                  class="ant-checkbox"
                >
                  <input
                    class="ant-checkbox-input"
                    name="Male"
                    type="checkbox"
                    value=""
                  />
                  <span
                    class="ant-checkbox-inner"
                  />
                </span>
              </label>
              <span
                class="ant-typography"
                style="margin-left: 1rem;"
              >
                Male
              </span>
            </div>
            <div
              class="input-group"
            >
              <label
                class="ant-checkbox-wrapper"
              >
                <span
                  class="ant-checkbox"
                >
                  <input
                    class="ant-checkbox-input"
                    name="Female"
                    type="checkbox"
                    value=""
                  />
                  <span
                    class="ant-checkbox-inner"
                  />
                </span>
              </label>
              <span
                class="ant-typography"
                style="margin-left: 1rem;"
              >
                Female
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-space-item"
      >
        <div
          class="ant-row demographics_gender-wrapper_selected-noof-patients"
        >
          <div
            class="ant-col"
          >
            <span
              class="ant-tag demographics_gender-wrapper_totalpatient-tag"
            >
              <svg>
                Vector-2.svg
              </svg>
              <span>
                 
                0
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-space ant-space-horizontal ant-space-align-center condition-container"
      style="border: 2px solid #d3d6d9; display: flex; flex-direction: column; margin-bottom: 0.5rem; padding: 20px 30px 12px 20px;"
    >
      <div
        class="ant-space-item"
        style="margin-right: 8px;"
      >
        <div
          class="ant-row demographics_agerange-wrapper"
          style="margin-left: -16px; margin-right: -16px;"
        >
          <div
            class="ant-col demographics_agerange-wrapper_drag-icon ant-col-sm-1 ant-col-md-2 ant-col-lg-1 ant-col-xl-1"
            style="padding-left: 16px; padding-right: 16px;"
          >
            <svg>
              Drag.svg
            </svg>
          </div>
          <div
            class="ant-col ant-col-sm-19 ant-col-md-17 ant-col-lg-14 ant-col-xl-14 ant-col-xxl-14"
            style="padding-left: 16px; padding-right: 16px;"
          >
            <div
              class="ant-row"
            >
              <div
                class="ant-col demographics_agerange-wrapper_Input-box ant-col-sm-8 ant-col-md-10 ant-col-lg-9 ant-col-xl-6 ant-col-xxl-3"
              >
                <div>
                  <input
                    class="ant-input"
                    name="min"
                    placeholder=""
                    style="border-radius: 0px; width: 100px;"
                    type="text"
                    value="0"
                  />
                </div>
              </div>
              <div
                class="ant-col demographics_agerange-wrapper_Input-box1"
              >
                <div>
                  <input
                    class="ant-input"
                    name="max"
                    placeholder=""
                    style="border-radius: 0px; width: 100px;"
                    type="text"
                    value="0"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-space-item"
      >
        <div
          class="ant-row demographics_agerange-wrapper_selected-noof-patients"
        >
          <div
            class="ant-col"
          >
            <span
              class="ant-tag demographics_agerange-wrapper_totalpatient-tag"
            >
              <svg>
                Vector-2.svg
              </svg>
              <span>
                 
                0
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
